weave/trace_server/sqlite_trace_server.py

```diff
@@ -736,3 +826,71 @@ def get_base_object_class(val: Any) -> Optional[str]:
                             elif "_class_name" in val:
                                 return val["_class_name"]
     return None
+
+
+def _quote_json_path(path: str) -> str:
```

Comment:
In my experience, end users will absolutely do things like have "file.ext" or a string "123" as keys in an object. A dotted notation is very convenient for the common case but we may want to consider also supporting a more verbose traversal format  like `object[1]["file.ext"]["42"]`

Reply:
Added a summary of limitations here: https://github.com/wandb/weave/pull/1656/commits/6f0e498dccd0c986280675ceff9d6293fd9b3a83 ... need to define this more crisply before opening up to public